nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmFacet.java

```diff
@@ -0,0 +1,75 @@
+package org.sonatype.repository.helm;
+
+import org.sonatype.nexus.repository.Facet;
+import org.sonatype.nexus.repository.FacetSupport;
+import org.sonatype.nexus.repository.storage.Asset;
+import org.sonatype.nexus.repository.storage.Bucket;
+import org.sonatype.nexus.repository.storage.Component;
+import org.sonatype.nexus.repository.storage.StorageTx;
+import org.sonatype.repository.helm.internal.HelmAssetAttributePopulator;
+import org.sonatype.repository.helm.internal.util.HelmDataAccess;
+
+import static org.sonatype.nexus.repository.storage.AssetEntityAdapter.P_ASSET_KIND;
+import static org.sonatype.repository.helm.internal.AssetKind.HELM_PACKAGE;
+
+public abstract class HelmFacet
```

Comment:
Let's make `HelmFacet` separate facet with implementation and just call its methods from hosted/proxy facets.